src/Scripting/Core/Hosting/AssemblyLoader/ShadowCopy.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.CodeAnalysis.Scripting.Hosting
     /// <summary>
     /// Represents a shadow copy of a single file.
     /// </summary>
-    public sealed class ShadowCopy
+    public sealed class FileShadowCopy
```

Comment:
I see the pattern, but I personally don't like either name.  Either can be used in isolation as a noun or adjective, and "File" is rarely an adjective, so I don't know which one is modifying which outside of the context of the method that returns this type.